### PR TITLE
cups: build with gssapi support

### PR DESCRIPTION
**Pull request by Hoshpak**
The cupsd.conf currently provided by the cups package contains a Kerberos
section. Since cupsd is built without gssapi support, the daemon fails
to verify it's configuration and refuses to start. This commit enables
building the package with gssapi support to fix this.

### CODE REVIEW

**Comment by xtraeme** [comment]
Well, I'd like to make this a build option instead.

**Comment by Hoshpak** [comment]
More like that? Normally I would have put it in configure_args but the position of the argument matters to the configure script and it doesn't pick it up otherwise. So I defined a new variable and appended it at the end of the configure arguments.

**Comment by Duncaen** [comment]
I would say, a `build_option` but disabled by default while fixing the config file.
No one requested this gssapi support and just adding it to make the default config file work seems a but useless.

File: srcpkgs/cups/template [hunk 1]
```diff
@@ -49,7 +58,7 @@ do_configure() {
 		--enable-libpaper --enable-avahi --enable-pam --enable-ssl \
 		--with-menudir=/usr/share/applications --with-xinetd=/etc/xinetd.d \
 		--with-optim="${CFLAGS}" --without-systemd \
-		--without-perl --without-php --without-python
+		--without-perl --without-php --without-python ${_extra_args}
```
Comment (Duncaen) [inline srcpkgs/cups/template]: Just use `$(vopt_enable gssapi)` instead of `${_extra_args}` here?

File: srcpkgs/cups/template [hunk 2]
```diff
@@ -23,11 +23,20 @@ make_dirs="
 	/var/spool/cups 	0755 cups lp
 	/etc/cups/ssl 		0700 cups lp"
 
+build_options="gssapi"
+desc_option_staging="Enable gssapi support"
```
Comment (Duncaen) [inline srcpkgs/cups/template]: `desc_option_staging` is wrong, but `desc_option_gssapi` is already defined in `common/options.description`.
Just remove it.

**Comment by Hoshpak** [comment]
I made the changes you requested. Building the package with gssapi support was the whole point of the pull request, the build option was added because @xtraeme requested it. So adding it just not to use it would be useless from my point of view. I see no harm in building cups with gssapi support, it is done this way by many distributions and is apparently now considered to be the default by upstream.

But I agree that the most important thing is to get cups working again so I don't want to argue to much about that. I've ammended the PR again to remove the option as default and pull in a patch taken from alpine to the cupsd.conf which removes the need to have the package built with gssapi enabled to start the daemon.

File: srcpkgs/cups/template [hunk 3]
```diff
@@ -23,11 +23,17 @@ make_dirs="
...
+
 # Package build options
 system_accounts="cups"
 cups_pgroup="lp"
 cups_homedir="/var/spool/cups"
 
+if [ ${build_option_gssapi} ]; then
+	makedepends+=" mit-krb5-devel"
+fi
+
```
Comment (Duncaen) [inline srcpkgs/cups/template]: Forgot about this, you can use `$(vopt_if gsapi mit-krb5-devel)` in `makdepends=`.

**Comment by Hoshpak** [comment]
done

**Comment by asergi** [comment]
Remove the old `--disable-gssapi` flag. Thanks.

**Comment by Hoshpak** [comment]
done. Thanks for pointing that out, I didn't notice it.

**Pull request merged**